docs/spec/supply/01_concepts.md

```diff
@@ -0,0 +1,39 @@
+# Concepts
+
+## Supply
+
+The `supply` module introduces a passive tracker for the `Supply` of coins in the chain, which allows to check for invariants of `Coins` with respect to the sum of `Coins` hold the stored `Accounts`.
+
+### Total Supply
+
+The total `Supply` of the network is equal to the sum of all coins from the `Account`s. The total supply is updated every time a `Coin` is minted (as part of the inflation mechanism) or burned (due to slashing or if a governance proposal is vetoed).
+
+## Module Accounts
+
+To keep track of the `Supply`, this module introduces a new type of `Account` used by other modules to control the flow of coins that come into and out of each of their modules. This design replaces the pools of coins that were stored on each of the modules, such as the `FeeCollectorKeeper` and the  staking `Pool`. The resoning of having this new `Account` type is to calculate the total supply without having to access each of the modules' pools of coins (previously stored in the `Store`), thus reducing the dependencies.
+
+The `ModuleAccount` interface is defined as follows:
+
+```go
+type ModuleAccount interface {
+  auth.Account            // same methods as the Account interface
+  GetName() string        // name of the module; used to obtain the address
+  GetPermission() string  // permission of module account (minter/burner/holder)
+}
+```
+
+The supply `Keeper` also introduces new wrapper functions for the `AccountKeeper` and the bank `Keeper` that are related to the `ModuleAccount`s in order to be able to:
+
+- Get and set `ModuleAccount`s by providing the `Name`.
+- Send coins from and to other `ModuleAccount`s or standard `Account`s (`BaseAccount` or `VestingAccount`) by passing only the `Name`.
+- `Mint` or `Burn` `Coins` for a `ModuleAccount` (restricted to its permissions).
+
+### Permissions
+
+Each `ModuleAccount` has a `Permission` that provides different object capabilities to perform certain actions. The avaliable permissions are:
+
+- `Holder`: is allowed to only hold and transfer `Coins` to other accounts.
```

Comment:
IMHO `holder` is not a good name for this permission. Need to think of alternatives...

Reply:
👍  let's move this comment to the supply PR